pkg/mcs/resourcemanager/server/manager.go

```diff
@@ -118,6 +83,28 @@ func (m *Manager) GetBasicServer() bs.Server {
 	return m.srv
 }
 
+// GetStorage returns the storage.
+func (m *Manager) GetStorage() endpoint.ResourceGroupStorage {
+	return m.storage
+}
+
+func (m *Manager) getOrCreateKeyspaceResourceGroupManager(keyspaceID uint32) *keyspaceResourceGroupManager {
+	krgm := m.getKeyspaceResourceGroupManager(keyspaceID)
+	if krgm == nil {
+		krgm = newKeyspaceResourceGroupManager(keyspaceID, m)
+		m.Lock()
+		m.krgms[keyspaceID] = krgm
+		m.Unlock()
+	}
+	return krgm
+}
+
```

Comment:
To avoid a race where two goroutines might both see `krgm == nil` and create it, should we move the m.Lock() call to before the initial if krgm == nil check?